lib/config/error-codes/throw.js

```diff
@@ -24,42 +24,73 @@ const { format } = require('util'),
 
-    const [domain, subdomain, errorName] = args;
-    const placeholders = args.length > 3 ? args.slice(3) : '';
-    const name = `${domain}-${subdomain}-${errorName}`;
-    const error = _.get(errorCodes, `${domain}.subdomains.${subdomain}.errors.${errorName}`);
-    if (!error) {
-      return module.exports.getError('internal', 'unexpected', 'unknown_error', ...placeholders);
-    }
-    const code =
-      (errorCodes[domain].code >> 24) |
-      (errorCodes[domain].subdomains[subdomain].code >> 16) |
-      errorCodes[domain].subdomains[subdomain].errors[errorName].code;
-    const message = format(error.message, ...placeholders);
-    if (error.class === 'PartialError') {
-      const body = args.slice(-1)[0];
-      return new errors[error.class](message, body, name, code);
-    }
-    return new errors[error.class](message, name, code);
-  },
+  if (! kuzzleError) {
+    return getError(
+      'internal',
+      'unexpected',
+      'unknown_error',
+      ...placeholders);
+  }
 
-  /** Manage errors to be thrown
-  *
-  * @param {...args} ...args
-  * @throws {KuzzleError}
-  */
+  const code =
+    (errorCodes[domain].code >> 24) |
+    (errorCodes[domain].subdomains[subdomain].code >> 16) |
+    errorCodes[domain].subdomains[subdomain].errors[error].code;
 
-  throw: (...args) => {
-    throw module.exports.getError(...args);
-  },
+  const message = format(kuzzleError.message, ...placeholders);
 
-  wrap: (domain, subdomain) => {
-    return {
-      getError: (...args) => module.exports.getError(domain, subdomain, ...args),
-      throw: (...args) => module.exports.throw(domain, subdomain, ...args)
-    };
+  if (kuzzleError.class === 'PartialError') {
+    const body = placeholders.slice(-1)[0];
+    return new errors[kuzzleError.class](message, body, errorName, code);
   }
+
+  return new errors[kuzzleError.class](message, errorName, code);
+}
+
+/**
+ * Construct and throw the corresponding error
+ *
+ * @param  {string} domain - Domain (eg: 'external')
+ * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
+ * @param  {string} error - Error name: (eg: 'index_not_found')
+ * @param  {...any} placeholders - Placeholders value to inject in error message
+ */
+function throwError (domain, subdomain, error, ...placeholders) {
+  throw getError(domain, subdomain, error, ...placeholders);
+}
+
+/**
+ * Wrap getError and throw with the provided domain and subdomain.
+
+ * The function have now the following signature:
+ *   - getError(error, ...placeholders)
+ *   - throw(error, ...placeholders)
+ *
+ * @param  {string} domain - Domain (eg: 'external')
+ * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
+ */
+function wrap (domain, subdomain) {
+  return {
+    getError: (error, ...placeholders) => getError(domain, subdomain, error, ...placeholders),
+    throw: (error, ...placeholders) => throwError(domain, subdomain, error, ...placeholders)
+  };
+}
+
+module.exports = {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Group all shorthand properties at either the beginning or end of this object declaration. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3499)